### PR TITLE
(RE-5776) Add missing pxp-agent directories

### PR DESCRIPTION
**Pull request by jcoconnor**
Also addresses PCP-57.
Adds:
  C:/ProgramData/PuppetLabs/pxp-agent/var
  C:/ProgramData/PuppetLabs/pxp-agent/var/spool
  C:/ProgramData/PuppetLabs/pxp-agent/var/run
  C:/ProgramData/PuppetLabs/pxp-agent/var/log
  C:/ProgramData/PuppetLabs/pxp-agent/etc
  C:/ProgramData/PuppetLabs/pxp-agent/etc/modules

### CODE REVIEW

**Comment by jcoconnor** [comment]
As per discussion with @richardc, adding all the directories specified in https://github.com/puppetlabs/puppet-specifications/blob/master/file_paths.md
Also need to add dummy components to force directories to install.

**Comment by richardc** [comment]
My apologies for not asking clearly enough before.  What's here looks good but we also need (at time of writing):

```
C:/ProgramData/PuppetLabs/pxp-agent/var/log
C:/ProgramData/PuppetLabs/pxp-agent/var/spool
```

**Comment by jcoconnor** [comment]
Ready for Review - updating header.

**Comment by richardc** [comment]
I'm not 100% on the WIX, but the paths look good to me. :+1:

File: wix/puppet.wxs [hunk 1]
```diff
@@ -386,6 +386,32 @@
...
+                </Component>
+              </Directory>
+              <Directory Id="pxpAgentDataFolderSpool" Name="spool" >
+                <Component Id="PXPAgentSpoolDir" Guid="F265A518-A7B7-4FA8-85D8-7306A10E1659" Permanent="yes" >
+                  <CreateFolder/>
+                </Component>
+              </Directory>
+            </Directory>
+            <Directory Id="pxpAgentDataFolderEtc" Name="etc">
+              <Directory Id="pxpAgentDataFolderModules" Name="modules" >
```
Comment (ferventcoder) [inline wix/puppet.wxs]: I thought this directory was going to be under program files?

Comment (ferventcoder) [inline wix/puppet.wxs]: Nevermind, I see this is for a config file.

File: wix/puppet.wxs [hunk 2]
```diff
@@ -386,6 +386,32 @@
               </Directory>
             </Directory>
           </Directory>
+          <Directory Id="pxpAgentDataFolder" Name="pxp-agent">
+            <Directory Id="pxpAgentDataFolderVar" Name="var">
+              <Directory Id="pxpAgentDataFolderLog" Name="log">
```
Comment (ferventcoder) [inline wix/puppet.wxs]: `log` isn't showing as being created in https://github.com/puppetlabs/puppet-specifications/blob/master/file_paths.md. I see a PR going up for that soon?

File: wix/puppet.wxs [hunk 3]
```diff
@@ -386,6 +386,32 @@
...
+                <Component Id="PXPAgentLogDir" Guid="DF0D0E2A-F069-4933-8AE9-5E779C8E93B9" Permanent="yes" >
+                  <CreateFolder/>
+                </Component>
+              </Directory>
+              <Directory Id="pxpAgentDataFolderRun" Name="run" >
+                <Component Id="PXPAgentRunDir" Guid="4AD77345-D8AE-40E4-B41C-706EC51FE63F" Permanent="yes" >
```
Comment (ferventcoder) [inline wix/puppet.wxs]: Looking closer, I don't see this as created either in the file_paths specs.

File: wix/puppet.wxs [hunk 4]
```diff
@@ -386,6 +386,32 @@
...
+                  <CreateFolder/>
+                </Component>
+              </Directory>
+              <Directory Id="pxpAgentDataFolderSpool" Name="spool" >
+                <Component Id="PXPAgentSpoolDir" Guid="F265A518-A7B7-4FA8-85D8-7306A10E1659" Permanent="yes" >
```
Comment (ferventcoder) [inline wix/puppet.wxs]: Or this directory.

**Comment by ferventcoder** [comment]
It appears the last time the file_paths was touched was https://github.com/puppetlabs/puppet-specifications/commit/823b317c439f6fe4d682cd1e74c134317a02f0f9. 

It appears we are now creating more folders than originally specified. Either we should stop trying to create so many folders or we need to update the file_path specification.

cc @kylog @Iristyle @richardc @parisiale @joshcooper (and anyone else I've forgotten here).

I'm usually off by a day of context so feel free to correct me on this one. :)

**Comment by ferventcoder** [comment]
Nevermind, I see we are now creating these directories. Told you I was off by a day off context!

File: wix/puppet.wxs (hunk 2 above)
Comment (ferventcoder) [inline wix/puppet.wxs]: https://github.com/puppetlabs/puppet-specifications/pull/56

File: wix/puppet.wxs (hunk 3 above)
Comment (ferventcoder) [inline wix/puppet.wxs]: https://github.com/puppetlabs/puppet-specifications/pull/56

File: wix/puppet.wxs (hunk 4 above)
Comment (ferventcoder) [inline wix/puppet.wxs]: https://github.com/puppetlabs/puppet-specifications/pull/56

File: wix/puppet.wxs (hunk 3 above)
Comment (ferventcoder) [inline wix/puppet.wxs]: puppetlabs/puppet-specifications#56

**Comment by ferventcoder** [comment]
Fixed up commit message and merged into master at https://github.com/puppetlabs/puppet_for_the_win/commit/0ddc156c8a1c80676c5fd269c4ee57998306a2de

**Pull request closed without merging**